enhancements/cloud-integration/aws/aws-sts-support.md

```diff
@@ -0,0 +1,171 @@
+
+#### Operator Changes
+
+All CVO managed operators (~5) that use CredentialsRequests will be updated to use the new `credentials` Secret key. Because this file can also contain a reference to a file with an AWS STS token, operators will then be able to build AWS clients without worrying about what type of authentication they are using.
+
+All CVO managed operators will also now mount in a [Kubernetes Projected Service Account Token](https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#service-account-token-volume-projection) to a consistent file location on disk. (/var/run/secrets/openshift/serviceaccount/token)
+
+```yaml
+      - name: bound-sa-token
+        projected:
+          defaultMode: 420
+          sources:
+          - serviceAccountToken:
+              audience: openshift
+              expirationSeconds: 3600
+              path: token
+```
+
+openshift-install will be modified to accept user provided service account signing keys and configure the Kubernetes APIServer to use them.
+
+For each `CredentialsRequest` the user will need to manually create the relevant Secret and role
+Users will have to manually provision the IAM roles, policies, and Secrets for each `CredentialsRequest`. This process will also require the name of the `ServiceAccount` that the pod consuming the credentials uses, which is typically created by the operator itself. However, we do not presently have the `ServiceAccount` name modeled in the `CredentialsRequest` API, and would like to defer doing this until we have more real world data and experience before making a permanent API commitment. As such, credentials will be documented as a list policy permissions, service account names, and target secrets, which the user will have to iterate:
+
+  1. Create an IAM Role with no policy attached, tagged with `kubernetes.io/cluster/[infraid]=owned` so it can be cleaned up on deprovision. (type = web identity, identity provider created by the admin manually)
+  1. Attach an in-line policy to the role which matches the permissions in the `CredentialsRequest` or documentation.
+  1. Ensure the trust relationship for the Role points to the cluster’s AWS IAM OIDC ARN with conditions to limit which `ServiceAccount` can assume this role.
+     ```json
+{
+	"Version": "2012-10-17",
+		"Statement": [
+		{
+			"Effect": "Allow",
+			"Principal": {
+				"Federated": "<OIDC_IDENTITY_PROVIDER_ARN>"
+			},
+			"Action": "sts:AssumeRoleWithWebIdentity",
+			"Condition": {
+				"StringEquals": {
+					"<OIDC_PROVIDER_URL_without_https://>:sub": "system:serviceaccount:<CredReq spec.Namespace>:<CredReq .spec.ServiceAccountName>"
```

Comment:
since we aren't modeling the SA name in the credReq, the namespace/name really is just the credReq Namespace and the Operator/Pod ServiceAccount